src/App.tsx

```diff
@@ -1,93 +1,123 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
+import { Status } from './types/filterENUM';
+import { useTodos } from './context/TodoContext';
+import { linkOptions } from './linkData';
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [filterStatus, setFilterStatus] = useState(Status.ALL);
+
+  const {
+    todos,
+    addTodo,
+    deleteCompletedTodos,
+    handleToggleAll,
+    incompletedTodosCount,
+    hasCompletedTodos,
+    filterTodos,
+  } = useTodos();
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleForSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.trim() === '') {
+      return;
+    }
+
+    addTodo(title);
+    setTitle('');
+  };
+
+  const handleClearCompleted = () => {
+    deleteCompletedTodos();
+  };
+
+  const handleFilterChange = (status: Status) => {
+    setFilterStatus(status);
+  };
```

Comment:
these functions are redundant
you may use `setFilterStatus` everywhere you use `handleFilterChange` and so on